volume-widget/volume.lua

```diff
@@ -186,15 +203,19 @@ local function worker(user_args)
     end
 
     function volume:inc(s)
-        spawn.easy_async(INC_VOLUME_CMD(device, s or step), function(stdout) update_graphic(volume.widget, stdout) end)
+        spawn.easy_async(INC_VOLUME_CMD(card, device, mixctrl, value_type, s or step), function(stdout) update_graphic(volume.widget, stdout) end)
     end
 
     function volume:dec(s)
-        spawn.easy_async(DEC_VOLUME_CMD(device, s or step), function(stdout) update_graphic(volume.widget, stdout) end)
+        spawn.easy_async(DEC_VOLUME_CMD(card, device, mixctrl, value_type, s or step), function(stdout) update_graphic(volume.widget, stdout) end)
     end
 
     function volume:toggle()
-        spawn.easy_async(TOG_VOLUME_CMD(device), function(stdout) update_graphic(volume.widget, stdout) end)
+        if toggle_cmd == nil then
```

Comment:
documentation done. i hope its understandable.

this looks a bit dirty to me. (spawn.easy_async takes some time i believe). the advantage of the current code is that the user doesn't need to take care, that the command doesn't output anything.

this is another approach:

```lua
spawn.easy_async(
    toggle_cmd == nil and TOG_VOLUME_CMD(card, device, mixctrl)
    or string.format([[sh -c "%s ; %s"]], toggle_cmd, GET_VOLUME_CMD(card, device, mixctrl)),
    function(stdout) update_graphic(volume.widget, stdout) end)
```